src/plugins/mer/meremulatormodedialog.cpp

```diff
@@ -80,6 +99,20 @@ QAction *MerEmulatorModeDialog::action() const
     return m_action;
 }
 
+MerEmulatorDevice::Ptr MerEmulatorModeDialog::emulator() const
+{
+    return m_emulator.toStrongRef();
+}
+
+QString MerEmulatorModeDialog::exec()
+{
+    execDialog();
+
+    return m_emulator.isNull()
+            ? QString()
+            : m_emulator.data()->deviceModel();
```

Comment:
I think we should be loud right here if exec is called without m_emulator. I would also prefer returning bool if the dialog was accepted instead of the model name (`execDialog()` needs to be updated for that). It will make the user code a bit more clear.

```
bool MerEmulatorModeDialog::exec()
{
    QTC_ASSERT(m_emulator != nullptr, return false);
    return execDialog();
}
```

Reply:
Done.